plugins/inputs/prometheus_scraper/metrics_type_handler.go

```diff
@@ -153,19 +139,24 @@ func isInternalMetric(metricName string) bool {
 	return false
 }
 
-// Decorate the Metrics with Metric Types
+// Decorate the Metrics with Metric Types.
+// Filter out Summary, Histogram and untyped Metrics and adding logging.
 func (mth *metricsTypeHandler) Handle(pmb PrometheusMetricBatch) (result PrometheusMetricBatch) {
-	// Filter out Summary, Histogram and untyped Metrics and adding logging
-	jobName, instanceId, err := GetScrapeTargetInfo(pmb)
-	if err != nil {
-		log.Printf("E! Failed to get Job Name and Instance ID from Prometheus metrics. \n")
-		return result
+	if len(pmb) == 0 {
+		log.Printf("D! Skip empty batch")
+		return nil
+	}
+
+	// All metrics in a batch are from same scrape target, we just need first one
+	jobName, instanceId := pmb[0].jobBeforeRelabel, pmb[0].instanceBeforeRelabel
+	if jobName == "" || instanceId == "" && len(pmb) != 0 {
```

Comment:
You don't need `len(pmb) != 0` as the check is already done at line 145

Reply:
🐶